packages/core/src/dashboards/component/AOPChart/AOPChart.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2024 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useEffect, useState} from 'react';
+import {Dimensions, View} from 'react-native';
+import {Chart, BarChart, LineChart, PieChart} from '@axelor/aos-mobile-ui';
+import {
+  fetchActionView,
+  fetchChartDataset,
+  fetchTypeChart,
+  getChartParameter,
+} from '../../../api/chart-api';
+
+interface AOPChartProps {
+  actionViewName: string;
+}
+
+const AOPChart = ({actionViewName}: AOPChartProps) => {
+  const [chart, setChart] = useState({type: '', dataset: [], title: ''});
+
+  useEffect(() => {
+    const fetchChartData = async () => {
+      const result = {type: '', dataset: [], title: ''};
+
+      try {
+        const actionViewResponse = await fetchActionView({
+          actionViewName: actionViewName,
+        });
+
+        const context = actionViewResponse?.data?.data[0]?.view?.context;
+        const chartName = actionViewResponse?.data?.data[0]?.view.views[0].name;
+
+        const typeResponse = await fetchTypeChart({chartName: chartName});
+        result.title = typeResponse?.data?.data?.title;
+        result.type = typeResponse?.data?.data?.series[0].type;
+        const onInit = typeResponse?.data?.data?.onInit;
+
+        let parameter = null;
+        if (onInit) {
+          const paramResponse = await getChartParameter({
+            chartName,
+            action: onInit,
+            context: context,
+          });
+          parameter = paramResponse?.data?.data[0].values;
+        }
+
+        const datasetResponse = await fetchChartDataset({
+          chartName,
+          parameter,
+          context,
+        });
+        result.dataset = datasetResponse?.data?.data?.dataset;
+      } catch (error) {
+        setChart({type: '', dataset: [], title: ''});
```

Comment:
I think we should reset `result` values here instead of calling setter as state value will be squashed when you exit the catch block